app/controllers/carts_controller.rb

```diff
@@ -0,0 +1,43 @@
+class CartsController < ApplicationController
+  include SessionsHelper
+
+  def show
+    @cart = Cart.find_by(user: current_user)
+  end
+
+  def add_item
+    cart = Cart.find_by(user: current_user)
+    if cart.present?
+      cart_item = cart.cart_items.find_by(product_id: params[:product_id])
+      if cart_item.present?
+        cart_item.quantity += params[:quantity].to_i
+        cart_item.save
+        redirect_to cart
+      else
+        #TODO:paramsの商品と個数を基にカートアイテムを作成
+        cart_item = cart.cart_items.new(product_id: params[:product_id], quantity: params[:quantity])
+        cart_item.save
+        redirect_to cart
+      end
+    else
+      #TODO:paramsの商品と個数を元にカートとカートアイテムを作成
+      @cart = Cart.new(user: current_user)
+      @cart.cart_items.new(product_id: params[:product_id], quantity: params[:quantity])
+      @cart.save
+      redirect_to cart
```

Comment:
この条件式の中ではcartはnilであり、情報を入れているのは@cartの変数なので、そちらを使用してください！
動作確認をしてみるとエラーになると思います。
```suggestion
      redirect_to @cart
```

Reply:
上記対応いたしました。
https://github.com/thimon21/railsthimon/pull/11/commits/64a20b5911324244130e832a0305b862e252d7af

Reply:
上記対応いたしました。
https://github.com/thimon21/railsthimon/pull/11/commits/8b58a333c9e2d1838e35d74508bc26ea4260ccc1